ua/variant.go

```diff
@@ -395,6 +404,11 @@ var errUnbalancedSlice = errors.New("unbalanced multi-dimensional array")
 // sliceDim determines the element type, dimensions and the total length
 // of a one or multi-dimensional slice.
 func sliceDim(v reflect.Value) (typ reflect.Type, dim []int32, count int32, err error) {
+	// null type
+	if v.Kind() == reflect.Invalid {
+		return nil, nil, 0, nil
+	}
+
 	// ByteString is its own type
 	if v.Type() == reflect.TypeOf([]byte{}) {
```

Comment:
`v.Type()` causes a panic when v is the zero value (`reflect.ValueOf(nil)`). We use the `Kind` method prior to checking `Type` here to handle zero value. See https://pkg.go.dev/reflect?tab=doc#Value.